project/Version.scala

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka
+
+import sbt._
+import sbt.Keys._
+
+/**
+ * Generate version.conf and akka/Version.scala files based on the version setting.
+ */
+object Version {
+
+  def versionSettings: Seq[Setting[_]] = inConfig(Compile)(Seq(
+    resourceGenerators <+= generateVersion(resourceManaged, _ / "akka-http-version.conf",
+      """|akka.http.version = "%s"
+         |"""),
+    sourceGenerators <+= generateVersion(sourceManaged, _ / "akka" / "http" / "Version.scala",
+      """|package akka.http
+         |
+         |import com.typesafe.config.Config
+         |
+         |object Version {
+         |  val current: String = "%s"
+         |  def check(config: Config): Unit = {
```

Comment:
If it's preferred to not have this code in a generated file I can move it to HttoExt. Also not sure if this check makes sense for Akka HTTP.